internal/controller/state_of_the_world.go

```diff
@@ -179,28 +183,76 @@ type BootOptionsBuilder struct {
 	isPrometheusOperatorInstalled bool
 }
 
-func (b *BootOptionsBuilder) getOptions() []controller.ControllerOption {
-	var opts []controller.ControllerOption
-	opts = append(opts, b.getGatewayAPIOptions()...)
-	opts = append(opts, b.getIstioOptions()...)
-	opts = append(opts, b.getEnvoyGatewayOptions()...)
-	opts = append(opts, b.getCertManagerOptions()...)
-	opts = append(opts, b.getConsolePluginOptions()...)
-	opts = append(opts, b.getDNSOperatorOptions()...)
-	opts = append(opts, b.getLimitadorOperatorOptions()...)
-	opts = append(opts, b.getAuthorinoOperatorOptions()...)
-	opts = append(opts, b.getObservabilityOptions()...)
-
-	return opts
+func (b *BootOptionsBuilder) getOptions() ([]controller.ControllerOption, error) {
+	var (
+		opts      []controller.ControllerOption
+		optionErr error
+	)
+
+	gwapiOpts, optionErr := b.getGatewayAPIOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, gwapiOpts...)
+
+	istioOpts, optionErr := b.getIstioOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, istioOpts...)
+
+	envoyGatewayOpts, optionErr := b.getEnvoyGatewayOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, envoyGatewayOpts...)
+
+	certManagerOpts, optionErr := b.getCertManagerOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, certManagerOpts...)
+
+	consoleOpts, optionErr := b.getConsolePluginOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, consoleOpts...)
+
+	dnsOpts, optionErr := b.getDNSOperatorOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, dnsOpts...)
+
+	limitadorOpts, optionErr := b.getLimitadorOperatorOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, limitadorOpts...)
+
+	authorinoOpts, optionErr := b.getAuthorinoOperatorOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, authorinoOpts...)
+
+	observabilityOpts, optionErr := b.getObservabilityOptions()
+	if optionErr != nil {
+		return opts, optionErr
+	}
+	opts = append(opts, observabilityOpts...)
+
+	return opts, nil
 }
 
-func (b *BootOptionsBuilder) getGatewayAPIOptions() []controller.ControllerOption {
+func (b *BootOptionsBuilder) getGatewayAPIOptions() ([]controller.ControllerOption, error) {
 	var opts []controller.ControllerOption
 	var err error
 	b.isGatewayAPIInstalled, err = kuadrantgatewayapi.IsGatewayAPIInstalled(b.manager.GetRESTMapper())
```

Comment:
nitpicking here, so feel free to leave as you implemented in the first place:

```go
b.isGatewayAPIInstalled, err = kuadrantgatewayapi.IsGatewayAPIInstalled(b.manager.GetRESTMapper())
if err != nil {
    return nil, err
}

if !b.isGatewayAPIInstalled {
    b.logger.Info("gateway api is not installed, skipping watches and reconcilers", "err", err)
    return opts, nil
}

```

Reply:
This is nice. I will add